src/ajax.js

```diff
@@ -444,80 +444,80 @@ jQuery.extend( {
 			// Default abort message
 			strAbort = "canceled",
 
-			// Fake xhr
-			jqXHR = {
-				readyState: 0,
+			// Fake xhr (ref the true Deferred instance above)
+			jqXHR = deferred;
```

Comment:
I think what @gibson042 is saying is that the jQXHR object returned right now does not have the ability to *resolve itself*, whereas if we returned a full `jQuery.Deferred` it would since the object would now have both `.resolve` and `.reject` methods. That violates the encapsulation we created. It also would violate the similar encapsulation provided by ES6 promises, which also cannot self-resolve.

Now you might say, "Who would do that?" but when you have millions of people writing jQuery, someone will do that (either on purpose or by accident) if you give them the ability.